eblob_kit.py

```diff
@@ -876,102 +1019,109 @@ def recover_index(data, destination):
+
+        for header in self._blob.data:
+            if not header:
+                logging.error('I have faced with broken record which I can not skip.')
+            if not header:
+                skipped_records += 1
+            elif header.flags.removed:
+                removed_records += 1
+            else:
+                copy_record(self._blob, blob, header)
+                copied_records += 1
+
+        logging.info('I have copied %s records, skipped %s and removed %s records',
+                     copied_records,
+                     skipped_records,
+                     removed_records)
 
     def copy_valid_records(self, destination):
         """Recover blob by copying only valid records from blob."""
-        basename = os.path.basename(self.blob.data.path)
+        basename = os.path.basename(self._blob.data.path)
         blob_path = os.path.join(destination, basename)
         blob = Blob.create(blob_path)
 
         copied_records = 0
         copied_size = 0
 
-        self.index_headers += self.data_recoverable_headers
+        self._index_headers += self.data_recoverable_headers
+        logging.info('Recovering blob %s -> %s', self._blob.data.path, blob_path)
 
-        with click.progressbar(iter(self.index_headers),
-                               length=len(self.index_headers),
-                               label='Recovering blob {} -> {}'
-                               .format(self.blob.data.path, blob_path)) as pbar:
-            for header in pbar:
-                copy_record(self.blob, blob, header)
-                copied_records += 1
-                copied_size += header.disk_size
-        click.echo('I have copied {} ({}) records {} -> {} '.format(
-            copied_records, sizeof_fmt(copied_size), self.blob.data.path, blob_path))
+        for header in self._index_headers:
+            copy_record(self._blob, blob, header)
+            copied_records += 1
+            copied_size += header.disk_size
+
+        logging.info('I have copied %s (%s) records %s -> %s ',
+                     copied_records,
+                     sizeof_fmt(copied_size),
+                     self._blob.data.path,
+                     blob_path)
 
     def fix(self, destination, noprompt):
-        """Check blob's data & index and try to fix them if they are broken."""
+        """Check blob's data & index and try to fix them if they are broken.
+
+        TODO(karapuz): remove all interactive user interaction.
+
+        """
         self.check(verify_csum=True, fast=False)
         self.print_check_report()
 
-        if self.valid:
+        if self._valid:
             return
 
-        if (not self.index_headers and
-                not self.index_removed_headers and
-                not self.index_uncommitted_headers):
+        if (not self._index_headers and
+            not self._stat.index_removed_headers and
+            not self._stat.index_uncommitted_headers):  # noqa: E129
```

Comment:
`# noqa: E129`?

Reply:
Yep, for example https://github.com/PyCQA/pycodestyle/issues/386,
to make _flake8_ shut up.

Reply:
Why not just use correct style?